src/languages.ts

```diff
@@ -0,0 +1,128 @@
+export const languages = [
+  "abap",
+  "aes",
+  "apex",
+  "azcli",
+  "bat",
+  "bicep",
+  "c",
+  "cameligo",
+  "clojure",
+  "coffeescript",
+  "cpp",
+  "csharp",
+  "csp",
+  "css",
+  "dart",
+  "dockerfile",
+  "ecl",
+  "elixir",
+  "flow9",
+  "fsharp",
+  "go",
+  "graphql",
+  "handlebars",
+  "hcl",
+  "html",
+  "ini",
+  "java",
+  "javascript",
+  "json",
+  "julia",
+  "kotlin",
+  "less",
+  "lexon",
+  "liquid",
+  "lua",
+  "m3",
+  "markdown",
+  "mips",
+  "msdax",
+  "mysql",
+  "objective-c",
+  "pascal",
+  "pascaligo",
+  "perl",
+  "pgsql",
+  "php",
+  "pla",
+  "plaintext",
+  "postiats",
+  "powerquery",
+  "powershell",
+  "proto",
+  "pug",
+  "python",
+  "qsharp",
+  "r",
+  "razor",
+  "redis",
+  "redshift",
+  "restructuredtext",
+  "ruby",
+  "rust",
+  "sb",
+  "scala",
+  "scheme",
+  "scss",
+  "shell",
+  "sol",
+  "sparql",
+  "sql",
+  "st",
+  "swift",
+  "systemverilog",
+  "tcl",
+  "twig",
+  "typescript",
+  "vb",
+  "verilog",
+  "xml",
+  "yaml"
+] as const;
+
+export type Language = typeof languages[number];
+
+/** The default extension of the known programming languages (to be used on download) */
+const defaultExtensions: {[key in Language]?: string} = {
+  typescript: "ts",
+  rust: "rs",
+  cpp: "cpp",
+  yaml: "yaml",
+  html: "html",
+};
+
+type LanguageExtensions = {[key in string]: Language};
+
+/** 
+ * Additional extensions that refer to known languages, to detect filetype on upload.
+ * NO NEED to add the default file extensions.
+ */
+const additionalExtensionsToLanguage: LanguageExtensions = {
+  "yml": "yaml",
+  "cc": "cpp",
+};
```

Comment:
It might be clearer to structure this as:

```ts
const languageExtensions: { Language: string[] } = {
  yaml: ["yml", "yaml"],
  cpp: ["cc", "cpp", "cxx", "C", "hpp", "hxx"],
  typescript: ["ts", "tsx"],
  // ...
};
```

I think we should also have a larger list of file extensions. Is it possible to take from VSCode's default list?

Reply:
I need a "default" extension per language for the download for sure. I wanted to keep this simple in case others want to add languages, but I guess this isn't really easier.

Another option to your suggestion would be

```ts
const languageExtensions: { Language: {defaultExtension: string, extensionAlias: string[] } } = {
  typescript: {
    defaultExtension: "ts",
    extensionAlias: ["tsx"],
  },
  //...
}
```

Also, do you have a reference to "VSCodes default list"? I'm not that familiar with monaco (aside as an enduser), is it monaco-editor\esm\vs\editor\editor.api.d.ts::languages.getLanguages what youre referring to ? Is the only thing I found on cursory scan, and I'm unsure if that works properly in our usecase, but I can try it for sure.